groups_cleanup/main.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+
+"""Removes group memberships matching a given regular expression."""
+
+from urllib.parse import urlencode
+import sys
+import re
+import googleapiclient.discovery
+
+CUSTOMER_ID = 'C010ys3gt'
+PAGE_SIZE = 1000  # How many results to fetch per request.
+
+
+def list_all_groups(service):
+    """Returns a dictionary of all groups in the domain, mapping addresses to group resource names."""
+    groups = {}
+    next_page_token = ''
+    while True:
+        search_query = urlencode(
+            {
+                'query': f'parent=="customerId/{CUSTOMER_ID}" && "cloudidentity.googleapis.com/groups.discussion_forum" in labels',
+                'page_size': PAGE_SIZE,
+                'page_token': next_page_token,
+            }
+        )
+        search_group_request = service.groups().search()
+        param = '&' + search_query
+        search_group_request.uri += param
+        response = search_group_request.execute()
+
+        if 'groups' not in response:
+            break
+
+        for group in response['groups']:
+            groups[group['groupKey']['id']] = group['name']
+
+        if 'nextPageToken' in response:
+            next_page_token = response['nextPageToken']
+        else:
+            next_page_token = ''
+
+        if len(next_page_token) == 0:
+            break
+
+    return groups
+
+
+def list_members(service, group_name):
+    """Returns a dictionary of all members in the given group, mapping addresses to member resource names."""
+    members = {}
+    next_page_token = ''
+    while True:
+        search_query = urlencode(
+            {
+                'page_size': PAGE_SIZE,
+                'page_token': next_page_token,
+            }
+        )
+        search_members_request = service.groups().memberships().list(parent=group_name)
+        param = '&' + search_query
+        search_members_request.uri += param
```

Comment:
Wow that's super gross haha

Reply:
Based on the official documentation...! :-/
https://cloud.google.com/identity/docs/how-to/create-google-groups#searching_for_a_google_group